tool/code_hygiene/analyze.py

```
@@ -0,0 +1,768 @@
+    command = [
+        git,
+        "-C",
+        str(repo),
+        "archive",
+        "--format=tar",
+        commit_hash,
+        "--",
+        *roots,
+    ]
+    try:
+        process = subprocess.Popen(
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)